benchmark.py

```diff
@@ -0,0 +1,211 @@
+import argparse
```

Comment:
This is helpful. Should we add this script to the devtools as a cli tool so others can also use it? And expand to other backend like vulkan as well.